### ISSUE TITLE
Feature proposal: recipe import/export/sync

### ISSUE THREAD

**Issue by lbischof** [grocy/grocy#247]
I manage my recipes in markdown and serve them via Hugo. I would like to keep my setup so that my recipes are publicly available, but would also love to have them available in Grocy. 

Ideas:
* Upload/sync new recipes via API.
* Serve all the recipes in yaml/json etc. at a certain url and Grocy scrapes it.
* Add export to markdown (with frontmatter data) option to Grocy.
* Add Git sync to Grocy.

**Comment by berrnd** [comment]
Have you checked the [API doc](https://demo-en.grocy.info/api)? The first two points are of course already possible using the API...

The integrated WYSIWYG editor ([Summernote](https://summernote.org/)) only supports HTML. I also like markdown, maybe adding an option to just disable the editor or even use another one which also supports markdown would be possible. But HTML/Markdown could also be easily converted in both directions...

"Syncing" a recipe including the ingredients and other options like included recipes would need a lot of checks & error handling, similar to #210. I see the benefit if it would be possible to add them from "any" source, but don't know if it's worth the effort to export a recipe from grocy to markdown, edit it there, then import it again - why not edit it directly (or use the API)...

There is also already a wish to be able to share recipes publicly via grocy - #186.

For Git sync - I think you mean versioning by this, don't see the real benefit of using Git for this, and would add a new system dependency, and a recipe is not only the text...

So in conclusion, I think this is something which should be done by using the API...

**Comment by lbischof** [comment]
Thanks for your reply. Am I missing something or is recipe not listed under Schemas?

**Comment by berrnd** [comment]
Seems so...

The API exposes under `/objects` just one to one the database tables.

So https://demo-en.grocy.info/api/objects/recipes will get all recipes (only those with `type` = `normal` are interesting, the others are internal ones, currently used for meal planning and are not displayed on the frontend).

https://demo-en.grocy.info/api/objects/recipes_pos will give alle the ingredients from all recipes, match them by `recipe_id` to the corresponding recipe.

https://demo-en.grocy.info/api/objects/recipes_nestings will give the the "included recipes".

**Comment by lbischof** [comment]
Thanks for the help. I will close this issue and see if I can write something with the API.

**Comment by barrelltitor** [comment]
Have you made anything like this already by any chance?
